core/src/main/java/org/elasticsearch/index/similarity/SimilarityService.java

```diff
@@ -35,11 +36,11 @@
 import java.util.function.BiFunction;
 
 public final class SimilarityService extends AbstractIndexComponent {
-
     public static final String DEFAULT_SIMILARITY = "BM25";
-    private final Similarity defaultSimilarity;
-    private final Similarity baseSimilarity;
-    private final Map<String, SimilarityProvider> similarities;
+
+    public static final Setting<Settings> SIMILARITY_SETTINGS = Setting.groupSetting("index.similarity.",
```

Comment:
Not sure how this would work. The concrete setting name and the acceptable inner settings can be resolved only when the user register a new similarity with a type like in:
`index.similarity.my_similarity.type: BM25`
This definition makes the settings `index.similarity.my_similarity.b`and `index.similarity.my_similarity.k1` acceptable. 
Are you saying that we should call `Setting#concreteSetting(name)` for every custom similarity that the user defines ? We would have to hack this a bit since the similarity `type` is the setting that would unlock the other possible settings.